opensrp-chw-anc/src/main/java/org/smartregister/chw/anc/util/JsonFormUtils.java

```diff
@@ -441,6 +470,86 @@ public static void updateFormField(JSONArray formFieldArrays, String formFieldKe
+                String fieldKey = valueField.optString(JsonFormConstants.KEY);
+                keysArrayList.add(fieldKey);
+            }
+
+            for (int k = 0; k < obs.size(); k++) {
+                Obs valueField = obs.get(k);
+                String fieldKey = valueField.getFormSubmissionField();
+                List<Object> values = (isNotNullOrEmpty(valueField.getHumanReadableValues())) ? valueField.getHumanReadableValues() : valueField.getValues();
+
+                if (isNotNullOrEmpty(values) && fieldKey.contains("_")) {
+                    fieldKey = fieldKey.substring(0, fieldKey.lastIndexOf("_"));
+                    if (keysArrayList.contains(fieldKey)) {
+                        String fieldValue = (String) values.get(0);
+                        if (StringUtils.isNotBlank(fieldValue)) {
+                            String fieldKeyId = valueField.getFormSubmissionField().substring(fieldKey.length() + 1);
+                            HashMap<String, String> hashMap = repeatingGroupMap.get(fieldKeyId) == null ? new HashMap<>() : repeatingGroupMap.get(fieldKeyId);
+                            hashMap.put(fieldKey, fieldValue);
+                            hashMap.put(Constants.JSON_FORM.REPEATING_GROUP_UNIQUE_ID, fieldKeyId);
+                            repeatingGroupMap.put(fieldKeyId, hashMap);
+                        }
+                    }
+                }
+            }
+        }
+        return repeatingGroupMap;
+    }
+
+    public static boolean isNotNullOrEmpty(List<Object> list) {
+        return list != null && !list.isEmpty();
+    }
+
+    /**
+     * Update the Pregnancy Outcome Event with formatted multiple babies' details Obs
+     *
+     * @param jsonString JsonForm String
+     * @param event      Pre-processed Pregnancy Outcome Event
+     */
+    public static void updatePregnancyOutcomeEventObs(String jsonString, Event event) throws JSONException {
+        JSONObject jsonFormObject;
+        try {
+            jsonFormObject = new JSONObject(jsonString);
+
+            LinkedHashMap<String, HashMap<String, String>> repeatingGroupsMap = buildRepeatingGroupMap(jsonFormObject, event.getObs(), Constants.JSON_FORM_KEY.NO_CHILDREN);
+            if (!repeatingGroupsMap.isEmpty()) {
+                JSONArray repeatArray = new JSONArray();
+                for (String key : repeatingGroupsMap.keySet()) {
+                    repeatArray.put(new JSONObject(Objects.requireNonNull(repeatingGroupsMap.get(key))));
+                }
+                List<Object> obsValues = new ArrayList<>();
+                obsValues.add(repeatArray.toString());
+                Obs formattedRepeatObs = new Obs();
+                formattedRepeatObs.setFieldCode("child_repeat_group_values_list");
+                formattedRepeatObs.setFieldDataType("repeatvalueslist");
+                formattedRepeatObs.setFieldType("concept");
+                formattedRepeatObs.setFormSubmissionField("repeatvalueslist");
```

Comment:
Move the string literals to constant class.

Reply:
Done ✅